### DISCUSSION TITLE
Other inputs than pure JSON

### DISCUSSION THREAD

**Discussion by jhoetter** [code-kern-ai/bricks#182] [Ideas]
I thought about chaining bricks, and I'd like to do things like

Voice Snippet -> Whisper bricks -> Hugging Face summarization brick -> emotion detection brick;

PDF scan -> OCR detection -> position-based invoice number detection bricks;

and so on. What I'm trying to say is that bricks could become even more useful if the data format is not only pure JSON, but we can also accept other formats. @LeonardPuettmann and @divyanshukatiyar we should look into this in the new year :)

**Comment by LeonardPuettmann** [comment]
I really like this. However, I am a bit worried about latency when chaining bricks modules. We should definitely try to keep things performant.

**Comment by jhoetter** [comment]
Why latency? :D Latency is something we can't really have impact on. I agree that we should try to keep things performant, but really don't see an issue with chaining bricks. You know how ETL pipelines basically are just chained commands, why not with bricks as well :)

**Reply by jhoetter** [reply]
I assume with latency you mean performance, right?
